config/class-site-details-index.php

```diff
@@ -0,0 +1,160 @@
+			static::$_instance = new Site_Details_Index();
+			static::$_instance->set_current_timestamp( $timestamp );
+			static::$_instance->init();
+		}
+
+		return static::$_instance;
+	}
+
+	public function init() {
+		add_filter( 'vip_site_details_index_data', [ $this, 'set_env_and_core' ] );
+	}
+
+	/**
+	 * Given the site details from the vip_site_details_index_data filter, set the environment and core values
+	 *
+	 * @param {array} $site_details The current site details
+	 * @return {array} The new site details
+	 */
+	public function set_env_and_core( $site_details ) {
+		global $wp_version;
+
+		$site_id = 0;
+		if ( defined( 'VIP_GO_APP_ID' ) && VIP_GO_APP_ID ) {
+			$site_id = VIP_GO_APP_ID;
+		} else {
+			if ( defined( 'FILES_CLIENT_SITE_ID' ) && FILES_CLIENT_SITE_ID ) {
+				$site_id = FILES_CLIENT_SITE_ID;
+			}
+		}
+
+		$environment_name = '';
+		if ( defined( 'VIP_GO_APP_ENVIRONMENT' ) && VIP_GO_APP_ENVIRONMENT ) {
+			$environment_name = strval( VIP_GO_APP_ENVIRONMENT );
+		} else {
+			if ( defined( 'VIP_GO_ENV' ) && VIP_GO_ENV ) {
+				$environment_name = strval( VIP_GO_ENV );
+			}
+		}
+
+		// log2logstash has it's own timestamp field as well. However, if/when we ever decide
+		// to swap indexes, we'll need our own. Added here so that the migration will be easier
+		// and all the fields we need are included in the actual site details index data.
+		$site_details['timestamp'] = $this->get_current_timestamp();
+		$site_details['client_site_id'] = $site_id;
+		$site_details['environment_id'] = $site_id;
+		$site_details['environment_name'] = $environment_name;
+		$site_details['plugins'] = $this->get_plugin_info();
+		$site_details['core']['wp_version'] = strval( $wp_version );
+		$site_details['core']['blog_id'] = get_current_blog_id();
+		$site_details['core']['is_multisite'] = is_multisite();
+		
+		return $site_details;
+	}
+
+	/**
+	 * Gather up all the information about all the plugins.
+	 *
+	 * Does not contain anything from vip-go-mu-plugins
+	 */
+	public function get_plugin_info() {
+		// Needed or get_plugins can't be found in some instances
+		require_once( ABSPATH . 'wp-admin/includes/plugin.php' );
+
+		$all_plugins = get_plugins();
+		$active_plugins = get_option( 'active_plugins' );
+		$plugins_enabled_via_code = wpcom_vip_get_filtered_loaded_plugins();
+
+		$plugin_info = array();
+
+		foreach ( $all_plugins as $key => $value ) {
+			$active = in_array( $key, $active_plugins, true );
+			$enabled_via_code = in_array( $key, $plugins_enabled_via_code, true );
+
+			$plugin_info[] = array(
+				'name' => $value['Name'],
+				'version' => $value['Version'],
+				'active' => $active,
+				'enabled_via_code' => $enabled_via_code,
+			);
+		}
+
+		return $plugin_info;
+	}
+
+	/**
+	 * Get the site details for the site the code is running on
+	 */
+	public function get_site_details() {
+		/* Filter that can be hooked on to provide information to the Site Details Index
```

Comment:
I really appreciate the investment in documenting this filter.

The [WPCS standard for documenting filters](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#4-hooks-actions-and-filters) is to use a DocBlock, rather than a multi-line comment. That's what WP-Parser and other compatible tools will look for.

```suggestion
		/**
		 * Filter that can be hooked on to provide information to the Site Details Index.
```